### PR TITLE
0.1.0 release

### LINKED ISSUE #55: 0.1.0 release and publication
The following list enumerates all tasks that need to be completed before:

* making the repo public,
* tagging a 0.1.0 release,
* publishing on PyPI/conda-forge/Spack, and
* publishing a paper on TorchGeo.

This list may change over time as we reevaluate the remaining tasks that need to be done.

### Documentation

- [x] Decide on a logo
- [x] Add tutorial (#38, #93)
- [x] Add installation instructions (#153)
- [x] Add contributing instructions
- [x] Document release procedure (https://github.com/microsoft/torchgeo/wiki/Releasing-Instructions)

### Continuous Integration

- [x] Set up codecov, add badge (must be public first?)
- [x] Set up RTD, add badge (must be public first?)
- [x] Determine minimum supported dependency versions (#32, #80)
- [x] Achieve 100% test coverage

### Publication

- [x] Decide on a publication venue (preprint on arXiv, paper in ICLR)
- [x] Benchmark different ways of performing the reprojection, etc. (#81)
- [x] Compare no-pretrain/imagenet-pretrain/torchgeo-pretrain
- [x] FInd publications that discuss and compare various geospatial datasets
- [x] Create an Overleaf document with ICLR template

### Datasets

- [x] Implement a few more GeoDatasets
  - [x] Chesapeake (#18)
  - [x] NAIP (#57)
  - [x] MSFT building footprints (#69)
- [x] Refactor indexing and getitem logic from datasets to GeoDataset base class (#73)

### Models

- [x] Set up infrastructure for hosting pre-trained model weights

### Samplers

- [x] Implement GridGeoSampler (#78)
- [x] More intelligent sampling (#82, #84)

### Trainers

- [x] Factor out task specific configuration logic into the different task classes
- [x] Pass the "task level" arguments to the DataModules as well as the Tasks. Batch size/seed/num workers should probably go in the "task level" arguments.
- [x] Add trainer for GeoDataset
- [x] Refactor trainers to reduce code duplication (#205)

### Transforms

- [x] Experiment with different ways of doing augmentation pipelines
- [x] Use Kornia, provide wrapper
- [x] Add geospatial-specific transforms (#127)

### PR DESCRIPTION
**Pull request by adamjstewart**
See https://github.com/microsoft/torchgeo/wiki/Releasing-Instructions

Closes #55

### CODE REVIEW

File: docs/user/installation.rst [hunk 1]
```diff
@@ -110,15 +110,3 @@ Optional dependencies can be installed by enabling build variants:
 Run ``spack info py-torchgeo`` for a complete list of variants.
 
 See the `spack documentation <https://spack.readthedocs.io/>`_ for more details.
-
-source
-------
-
-TorchGeo can also be installed from source using the ``setup.py`` file and setuptools.
```
Comment (adamjstewart) [inline docs/user/installation.rst:117]: Direct invocation of `setup.py` is deprecated: https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html

File: conf/task_defaults/bigearthnet.yaml [hunk 2]
```diff
@@ -9,8 +9,8 @@ experiment:
     in_channels: 14
```
Comment (adamjstewart) [inline conf/task_defaults/bigearthnet.yaml:9]: From what I can tell, `conf/task_defaults/*.yaml` are only used during testing. We should consider moving them to `tests/conf`

Comment (calebrob6) [reply conf/task_defaults/bigearthnet.yaml:9]: They are used here -- https://github.com/microsoft/torchgeo/blob/main/train.py#L97. They provide the default arguments for each trainer.

Comment (calebrob6) [reply conf/task_defaults/bigearthnet.yaml:9]: See https://github.com/microsoft/torchgeo/pull/45

Comment (calebrob6) [reply conf/task_defaults/bigearthnet.yaml:9]: I think it makes sense to re-do the configs -- I'll open an issue describing this

File: tests/test_train.py [hunk 3]
```diff
@@ -134,20 +134,34 @@ def test_config_file(tmp_path: Path) -> None:
...
         sys.executable,
         "train.py",
         "experiment.name=test",
         "program.output_dir=" + str(output_dir),
-        "program.data_dir=" + data_dir,
         "program.log_dir=" + str(log_dir),
         "trainer.fast_dev_run=1",
         "experiment.task=" + task,
         "program.overwrite=True",
+        "config_file=" + os.path.join("conf", "task_defaults", task + ".yaml"),
```
Comment (adamjstewart) [inline tests/test_train.py:165]: @calebrob6 without this line, the default configs aren't being loaded, so there might be something broken with default loading.

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -17,9 +17,19 @@ jobs:
...
         python-version: 3.9
+    - name: Install apt dependencies
+      run: |
+        sudo apt-add-repository ppa:ubuntugis/ubuntugis-unstable
+        sudo apt-get update
+        sudo apt-get install gdal-bin libgdal-dev
     - name: Install pip dependencies
-      run: pip install .[tests]
+      run: |
+        pip install gdal tqdm  # TODO: these deps shouldn't be needed
```
Comment (adamjstewart) [inline .github/workflows/release.yaml:27]: @isaaccorley the indices tutorial imports things that aren't TorchGeo deps, does it need to?

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -17,9 +17,19 @@ jobs:
...
+        pip install .[datasets,tests,train]
+        pip install -r docs/requirements.txt
     - name: Run notebook checks
+      env:
+        MLHUB_API_KEY: ${{ secrets.MLHUB_API_KEY }}
```
Comment (adamjstewart) [inline .github/workflows/release.yaml:32]: Alternatively, we could use a different dataset that doesn't require an API key to download

File: torchgeo/trainers/byol.py [hunk 6]
```diff
@@ -451,3 +451,6 @@ def validation_step(  # type: ignore[override]
         )
 
         self.log("val_loss", loss, on_step=False, on_epoch=True)
+
+    def test_step(self, *args: Any) -> None:  # type: ignore[override]
+        """No-op, does nothing."""
```
Comment (adamjstewart) [inline torchgeo/trainers/byol.py:456]: PyTorch Lightning 1.5 added a feature to check the validity of a trainer. If you pass a DataModule with a `test_loader` but your trainer doesn't have a `test_step`, it raises an error.

File: tests/test_train.py (hunk 3 above)
Comment (calebrob6) [reply tests/test_train.py:165]: The default config is loaded by name here -- https://github.com/microsoft/torchgeo/blob/main/train.py#L97.

Comment (adamjstewart) [reply tests/test_train.py:165]: Yes, but if you comment out this line, you'll see that it isn't loading the default config.

Comment (calebrob6) [reply tests/test_train.py:165]: It definitely is loading the default config, else it would throw an error about not finding the default config

Comment (calebrob6) [reply tests/test_train.py:165]: Because of the way configurations are merged, `experiment.datamodule.root_dir` was being overridden by the defaults in `defaults.conf`

Comment (adamjstewart) [reply tests/test_train.py:165]: Yeah, I realize I've been abusing the config system by modifying it to suit the tests. What if we:

1. Keep doing this until we get the release out since the `conf` directory won't get installed anyway
2. After the release is out, create a `tests/conf` directory for test-specific configuration settings
3. Think about #227 post-rebuttal when we have more time

We could also do 2 real quick, it wouldn't take that long.

Comment (calebrob6) [reply tests/test_train.py:165]: Can you make a TODO here or somewhere that we'll remember?

Comment (adamjstewart) [reply tests/test_train.py:165]: I'll create an issue so we don't re-trigger the tests

File: .github/workflows/release.yaml (hunk 4 above)
Comment (isaaccorley) [reply .github/workflows/release.yaml:27]: I can revamp them but requiring that notebooks only use torchgeo deps might be too restrictive.

Comment (calebrob6) [reply .github/workflows/release.yaml:27]: The code definitely uses these two libraries.
- tqdm is simple to include as a dependency, it will be indirectly included anyway, and is quite useful in notebooks and command line scripts
- The gdal bit can be replaced quite easily with rasterio

File: .github/workflows/release.yaml (hunk 5 above)
Comment (calebrob6) [reply .github/workflows/release.yaml:32]: I'd rather not have an API key secret, and would rather use UCMerced instead of Cyclone. Lets put that on the 0.2 release path.

File: .github/workflows/release.yaml (hunk 4 above)
Comment (adamjstewart) [reply .github/workflows/release.yaml:27]: Agreed, let's do this for the next release

**Review by calebrob6 — APPROVED** [review]
